samples/snippets/csharp/how-to/strings/CompareStrings.cs

```diff
@@ -0,0 +1,255 @@
+                }
+            }
+            //</snippet6>
+        }
+
+        private static void SortListOfStrings()
+        {
+            // <Snippet7>
+            List<string> lines = new List<string>
+            {
+                @"c:\public\textfile.txt",
+                @"c:\public\textFile.TXT",
+                @"c:\public\Text.txt",
+                @"c:\public\testfile2.txt"
+            };
+
+            Console.WriteLine("Non-sorted order:");
+            foreach (string s in lines)
+            {
+                Console.WriteLine($"   {s}");
+            }
+
+            Console.WriteLine("\n\rSorted order:");
+
+            lines.Sort((left, right) => left.CompareTo(right));
+            foreach (string s in lines)
+            {
+                Console.WriteLine($"   {s}");
+            }
+            // </Snippet7>
+        }
+
+        private static void SearchSortedList()
+        {
+            // <Snippet8>
+            List<string> lines = new List<string>
+            {
+                @"c:\public\textfile.txt",
+                @"c:\public\textFile.TXT",
+                @"c:\public\Text.txt",
+                @"c:\public\testfile2.txt"
+            };
+            lines.Sort((left, right) => left.CompareTo(right));
+ 
+            string searchString = @"c:\public\TEXTFILE.TXT";
+            Console.WriteLine($"Binary search for <{searchString}>");
+            int result = lines.BinarySearch(searchString, null);
+            ShowWhere<string>(lines, result);
+
+            Console.WriteLine($"{(result > 0 ? "Found" : "Did not find")} {searchString}");
+
+            void ShowWhere<T>(IList<T> collection, int index)
+            {
+                if (index < 0)
+                {
+                    index = ~index;
+
+                    Console.Write("Not found. Sorts between: ");
+
+                    if (index == 0)
+                        Console.Write("beginning of array and ");
+                    else
+                        Console.Write($"{collection[index - 1]} and ");
+
+                    if (index == collection.Count)
+                        Console.WriteLine("end of array.");
+                    else
+                        Console.WriteLine($"{collection[index]}.");
+                }
+                else
+                {
+                    Console.WriteLine($"Found at index {index}.");
+                }
+            }
+            //</snippet8>
+        }
+
+        private static void CompareAcrossCultures()
+        {
+            //<snippet9>  
+            string first = "Sie tanzen auf der Straße.";
+            string second = "Sie tanzen auf der Strasse.";
+
+            Console.WriteLine($"First sentence is <{first}>");
+            Console.WriteLine($"Second sentence is <{second}>");
+
+            System.Threading.Thread thread = System.Threading.Thread.CurrentThread;
+            System.Globalization.CultureInfo originalCulture = thread.CurrentCulture;
```

Comment:
I am not sure if this code depends on specific version of the framework. if this for net core or after 4.6 in the full framework, I would write this simply as

 System.Globalization.CultureInfo originalCulture = CultureInfo.CurrentCulture;

and setting current culture would be 

CultureInfo.CurrentCulture = ew System.Globalization.CultureInfo("en-US");